drivers/syslog/ramlog.c

```diff
@@ -286,84 +296,42 @@ static int ramlog_addchar(FAR struct ramlog_dev_s *priv, char ch)
 
   flags = enter_critical_section();
 
-  /* Calculate the write index AFTER the next byte is written */
-
-  nexthead = priv->rl_head + 1;
-  if (nexthead >= priv->rl_bufsize)
-    {
-      nexthead = 0;
-    }
+  /* Calculate the size in the buffer */
 
-  /* Would the next write overflow the circular buffer? */
+  space = (int)priv->rl_head - (int)priv->rl_tail >= 0 ?
+          priv->rl_bufsize - (priv->rl_head - priv->rl_tail) - 1:
+          priv->rl_tail - priv->rl_head - 1;
 
-  if (nexthead == priv->rl_tail)
+  if (len > space)
```

Comment:
need handle len > rl_bufsize correctly

Reply:
len > rl_bufsize already included in len > space

Reply:
you need additional handle in case of len > rl_bufsize, otherwise the buffer overlofw will happen.

Reply:
You are right, there will be a problem when len>2*rl_bufsize